plugin/src/main/java/io/jenkins/plugins/casc/core/LabelAtomConfigurator.java

```diff
@@ -0,0 +1,48 @@
+package io.jenkins.plugins.casc.core;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Label;
+import hudson.model.labels.LabelAtom;
+import hudson.model.labels.LabelAtomProperty;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.impl.attributes.DescribableListAttribute;
+import io.jenkins.plugins.casc.impl.attributes.MultivaluedAttribute;
+import io.jenkins.plugins.casc.model.Mapping;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import static io.jenkins.plugins.casc.Attribute.noop;
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class LabelAtomConfigurator extends BaseConfigurator<LabelAtom> {
+
+    @Override
+    public Class<LabelAtom> getTarget() {
+        return LabelAtom.class;
+    }
+
+    @Override
+    protected LabelAtom instance(Mapping mapping, ConfigurationContext context) throws ConfiguratorException {
+        return new LabelAtom(mapping.getScalarValue("name"));
+    }
+
+    @NonNull
+    @Override
+    public Set<Attribute<LabelAtom, ?>> describe() {
+        return new HashSet<>(Arrays.asList(
+            new Attribute<LabelAtom, String>("name", String.class)
+                .getter(Label::getName)
+                .setter( noop() ),
+            new MultivaluedAttribute<LabelAtom, LabelAtomProperty>("properties", LabelAtomProperty.class)
```

Comment:
you can create your object that you can use instead if that makes sense?
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java#L45

i.e. UserWithPassword,

I haven't looked closely into this though

Reply:
But how can I create my object if I don't know what each implementation provides? One LabelAtomProperty might have a couple of String fields, another one might have a String and an integer for example.

Reply:
Does it work if the implementation LabelAtomProperty has a databoundcontructor on it?

i.e. maybe it just needs adding in a few places

Reply:
No, you can see how "TestProperty" has a DataboundConstructor and it's not working.